kmath-functions/src/commonMain/kotlin/space/kscience/kmath/functions/Polynomial.kt

```diff
@@ -5,128 +5,498 @@
 
 package space.kscience.kmath.functions
 
-import space.kscience.kmath.misc.UnstableKMathAPI
-import space.kscience.kmath.operations.*
-import kotlin.contracts.InvocationKind
-import kotlin.contracts.contract
-import kotlin.math.max
-import kotlin.math.pow
+import space.kscience.kmath.operations.Ring
+import space.kscience.kmath.operations.invoke
+import kotlin.js.JsName
+import kotlin.jvm.JvmName
 
-/**
- * Polynomial coefficients model without fixation on specific context they are applied to.
- *
- * @param coefficients constant is the leftmost coefficient.
- */
-public class Polynomial<out T>(public val coefficients: List<T>) {
-    override fun toString(): String = "Polynomial$coefficients"
-}
 
 /**
- * Returns a [Polynomial] instance with given [coefficients].
+ * Abstraction of polynomials.
  */
-@Suppress("FunctionName")
-public fun <T> Polynomial(vararg coefficients: T): Polynomial<T> = Polynomial(coefficients.toList())
+public interface Polynomial<C>
```

Comment:
> This abstraction is useless.

Yeah, I agree that this interface is useless and should be removed (or replaced).

> It should at least provide a way to extract individual coefficients. Maybe expose coefficients as `Buffer`

In general multivariate case I don't see a way to expose them, because in that case we should either declare way to represent term with both its signature and its coefficient or declare common order on monomials (independently on number of variables!).

It is possible in univariate case by just representing the polynomial as a sequence. But I don't know if it is necessary.